### PR TITLE
implement identification and authentication interface

### PR DESCRIPTION
**Pull request by kianooshaz**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,3 +1,3 @@
-module github.com/persiangophers/userland
+module github.com/kianooshaz/userland
```
Comment (nasermirzaei89) [inline go.mod:1]: module ro chera taghir dadi?

File: main.go [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package main
+
+func main() {
+
```
Comment (nasermirzaei89) [inline main.go:4]: Not complete

File: internal/service/service.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
+package contract
+
+import (
+	"github.com/kianooshaz/userland/internal/dto"
+	"github.com/kianooshaz/userland/internal/entity"
+)
+
+type (
+	Ident interface {
```
Comment (nasermirzaei89) [inline internal/service/service.go:9]: in kholase kardane na khana karde

File: internal/service/service.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+		GetUserWithID(id int) (*entity.User, error)
+		GetUserWithUUID(uuid string) (*entity.User, error)
+		GetUserWithUsername(username string) (*entity.User, error)
+		GetUserWithPhonenumber(phonenumber string) (*entity.User, error)
+		GetUserWithEmail(email string) (*entity.User, error)
+	}
+
+	Auth interface {
+		AddUser(req *dto.CreateUserRequest) error
```
Comment (nasermirzaei89) [inline internal/service/service.go:20]: I think AddUser is not responsibility of Auth

File: internal/dto/user.go [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package dto
+
+type (
+	CreateUserRequest struct {
+		Firstname   string `json:"firstname"`
+		Lastname    string `json:"lastname"`
+		Username    string `json:"username"`
+		Email       string `json:"email"`
+		Phonenumber string `json:"phonenumber"`
```
Comment (nasermirzaei89) [inline internal/dto/user.go:9]: use `PhoneNumber` instead of `Phonenumber`

File: internal/entity/user.go [hunk 6]
```diff
@@ -0,0 +1,22 @@
+package entity
+
+import "time"
+
+type (
+	User struct {
+		ID            int        `json:"id"`
```
Comment (nasermirzaei89) [inline internal/entity/user.go:7]: why id is int? is it postgres id?

File: internal/service/service.go [hunk 7]
```diff
@@ -0,0 +1,38 @@
+package contract
```
Comment (nasermirzaei89) [inline internal/service/service.go:1]: why package name is not as the same as directory?

**Review by nasermirzaei89 — CHANGES_REQUESTED** [review]

File: internal/entity/user.go (hunk 6 above)
Comment (sina-devel) [reply internal/entity/user.go:7]: anyway, ID is a positive integer we can use uint.

**Pull request closed without merging**